Tests/Common/SimulatedDevice.cs

```diff
@@ -184,128 +185,105 @@ public LoRaPayloadData CreateConfirmedDataUpMessage(string data, uint? fcnt = nu
-        }
+                var appSKey = OTAAKeysGenerator.CalculateAppSessionKey(appNonce, netid, devNonce, LoRaDevice.AppKey.Value);
+                var nwkSKey = OTAAKeysGenerator.CalculateNetworkSessionKey(appNonce, netid, devNonce, LoRaDevice.AppKey.Value);
+                var devAddr = payload.DevAddr;
 
-        // Sends confirmed message
-        public async Task SendConfirmedMessageAsync(SimulatedPacketForwarder simulatedPacketForwarder, string payload)
-        {
-            var token = await RandomTokenGenerator.GetTokenAsync();
-            if (LastPayload == null)
-                LastPayload = new PhysicalPayload(token, PhysicalIdentifier.PushData, null);
-            var header = LastPayload.GetSyncHeader(simulatedPacketForwarder.MacAddress.ToArray());
+                // if mic check failed, return false
+                /*
+                if (!payload.CheckMic(BitConverter.ToString(nwkSKey).Replace("-", "")))
+                {
+                    return false;
+                }
+                */
 
-            var confirmedMessage = CreateConfirmedDataUpMessage(payload);
-            LastPayload = await simulatedPacketForwarder.SendAsync(header, confirmedMessage.GetByteMessage());
+                LoRaDevice.AppSKey = appSKey;
+                LoRaDevice.NwkSKey = nwkSKey;
+                NetId = BitConverter.ToString(netid).Replace("-", string.Empty, StringComparison.Ordinal);
+                AppNonce = BitConverter.ToString(appNonce).Replace("-", string.Empty, StringComparison.Ordinal);
+                LoRaDevice.DevAddr = BitConverter.ToString(devAddr.ToArray()).Replace("-", string.Empty, StringComparison.Ordinal);
 
-            TestLogger.Log($"[{LoRaDevice.DeviceID}] Confirmed data: {BitConverter.ToString(header).Replace("-", string.Empty, StringComparison.Ordinal)} {payload}");
+                return true;
+            }
+            catch (Exception ex)
+            {
+                TestLogger.Log("ERROR:" + ex);
+            }
 
-            // TestLogger.Log($"[{LoRaDevice.DevAddr}] Sending data: {BitConverter.ToString(header).Replace("-", "")}{Encoding.UTF8.GetString(gatewayInfo)}");
+            return false;
         }
 
         // Performs join
-        // THE FOLLOWING METHODS ARE LEFT HERE FOR REFERENCE
-        // THEY ACT AS DOCUMENTATION UNTIL WE MIGRATE THE SIMULATOR
-#pragma warning disable CA1822 // Mark members as static
-#pragma warning disable IDE0060 // Remove unused parameter
-        public async Task<bool> JoinAsync(SimulatedPacketForwarder packetForwarder, int timeoutInMs = 30 * 1000)
-#pragma warning restore IDE0060 // Remove unused parameter
-#pragma warning restore CA1822 // Mark members as static
+        public async Task<bool> JoinAsync(LoRaRequest joinRequest, SimulatedBasicsStation basicsStation, int timeoutInMs = 30 * 1000)
```

Comment:
I would advocate to use `TimeSpan? timeout = null` as a parameter and then initialize it as needed, e.g. as `timeout ??= TimeSpan.FromSeconds(30)`. This would make it easier to not need to keep track of the units for the timeouts.